redaxo/src/core/functions/function_rex_escape.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+/**
+ * Escapes a variable.
+ *
+ * This function is adapted from code coming from Twig.
+ * (c) Fabien Potencier
+ * https://github.com/twigphp/Twig/blob/103cae817d68b56ddcb50c051a6ed7980d746409/lib/Twig/Extension/Core.php#L880-L1106
+ *
+ * @package redaxo\core
+ *
+ * @param mixed  $value    The value to escape
+ * @param string $strategy One of "html", "html_attr", "css", "js", "url"
+ *
+ * @throws InvalidArgumentException
+ *
+ * @return mixed
+ */
+function rex_escape($value, $strategy = 'html')
+{
+    if (!is_string($value)) {
+        if (is_array($value)) {
+            foreach ($value as $k => $v) {
+                $value[$k] = rex_escape($v, $strategy);
+            }
+
+            return $value;
+        }
+
+        if ($value instanceof \stdClass) {
+            foreach (get_object_vars($value) as $k => $v) {
+                $value->$k = rex_escape($v, $strategy);
+            }
```

Comment:
In den Fragmenten hatten wir das allgemein für Objekte. Ich finde das macht so aber nur Sinn für Array-artige Objekte.
Insbesondere Objekte mit `__toString()`-Methode werden nun wie Strings behandelt.
Ich denke, das passt so besser.